packages/native/src/components/Form/Input/BaseInput/index.tsx

```diff
@@ -83,9 +104,33 @@ export const InputRenderRightContainer = styled(FlexBox).attrs(() => ({
   pr: "16px",
 }))``;
 
-export default function Input(props: InputProps): JSX.Element {
-  const { value, disabled, error, renderLeft, renderRight, ...textInputProps } =
-    props;
+// Yes, this is dirty. If you can figure out a better way please change the code :).
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const IDENTITY = (_: any): any => _;
+
+export default function Input<T = string>(props: InputProps<T>): JSX.Element {
+  const {
+    value,
+    onChange,
+    onChangeText,
+    disabled,
+    error,
+    renderLeft,
+    renderRight,
+    serialize = IDENTITY,
+    deserialize = IDENTITY,
+    ...textInputProps
+  } = props;
+
+  const inputValue = useMemo(() => serialize(value), [serialize, value]);
+
+  const handleChange = useCallback(
+    (value: string) => {
+      onChange(deserialize(value));
```

Comment:
onChange on desktop return the value from the input event. There it return the native input event unchanged.
Maybe keep the same behavior for the 2 platforms ?

Also it weird to have onChange mandatory when you only want to use onChangeText